addons/dropdown-costume-numbers/userscript.js

```diff
@@ -0,0 +1,39 @@
+export default async function ({ addon, console }) {
+  // The boolean value is for whether the last three items should not have a number - e.g. for the "next background", "previous background" etc
+  const COSTUME_BLOCKS = {
+    looks_switchcostumeto: false,
+    looks_switchbackdropto: true,
+    looks_switchbackdroptoandwait: true,
+  };
+
+  const dropDownDiv = await addon.tab.waitForElement(".blocklyDropDownDiv");
+  const workspace = await addon.tab.traps.getWorkspace();
+
+  // This will show the original text without the numbers for a bit.
+  // I'm not aware of any better way to detect dropdown close, though
+  const waitForDropdownClose = () => {
+    const currentStyle = dropDownDiv.getAttribute("style");
+    return new Promise((resolve) =>
+      setInterval(() => {
```

Comment:
Looks like we're looking for a way to run code only when there's a new dropdown that should be affected. Of course, a setInterval would do the trick, but it's pretty inefficient. Using waitForElement can also hurt performance depending on how it's used.

The best way to avoid setInterval (or other ugly ways to check if there's a new dropdown) is probably taking inspiration from the editor-searchable-dropdown addon:

Instead of while(true) waitForElement etc...
```js
const Blockly = await addon.tab.traps.getBlockly();
const oldDropDownDivShow = Blockly.DropDownDiv.show;
Blockly.DropDownDiv.show = function (...args) {
  try {
    // there's probably a new dropdown in the page
    // do stuff with the dropdown normally here...
  } catch(e) {console.error(e);}
  return oldDropDownDivShow.call(this, ...args);
}

Reply:
I think it would make more sense to override `Blockly.FieldDropdown.prototype.showEditor_` in this case. This would also make checking the dropdown type easier (`this.name === "COSTUME"`, `this.name === "BACKDROP"`).